kano_qt_apps/lib/src/app_list.cpp

```diff
@@ -59,7 +58,10 @@ void AppList::add_app(JSON_Object *new_app, std::shared_ptr<App> fallback)
 
 void AppList::add_app_from_file(std::string file_path, std::shared_ptr<App> fallback)
 {
-    this->add_app(App(file_path), std::move(fallback));
+    App app=App(file_path);
+    if (app.get_hidden() == false) {
+        this->add_app(app, std::move(fallback));
+    }
```

Comment:
I think that you want to move this into the data presentation rather than the loading - an app should still be stored in this object even if it isn't going to be displayed as this is used. As such I think that you want to override the [`QAppList::get_app_list()`](https://github.com/KanoComputing/kano-apps/blob/rc/kano_qt_apps/lib/src/q_app_list.cpp#L62-L74) function in the [`InstalledAppList`](https://github.com/KanoComputing/kano-apps/blob/rc/kano_qt_apps/lib/includes/installed_app_list.h) class (subclass of [`QAppList`](https://github.com/KanoComputing/kano-apps/blob/rc/kano_qt_apps/lib/includes/q_app_list.h))

Also,

```c++
if (app.get_hidden() == false) {
   ...
}
```

is just

```c++
if (!app.get_hidden()) {
   ...
}
```